src/pytest_workflow/plugin.py

```diff
@@ -222,6 +304,9 @@ def collect(self):
         if not (set(self.config.getoption("workflow_tags")
                     ).issubset(set(self.tags))):
             return []
+        else:
```

Comment:
Meta-comment.

So, we create a temporary directory at instantiation of each workflow object, regardless of whether it's run. We then also have to remove output of workflows that would never have gotten any output in the first place. That seems a little strange to me.

What about doing the temporary directory creation in the collect step? That would also obviate the need for keeping a `self.config.workflow_cleanup_dirs` variable around, as the directories to be removed can be looked up using an instance field or property from the workflows that were run. 

E.g. something like:

```python
class YamlFile(pytest.File):
    def __init__(args, kwargs):
        self.tempdir = None
    def collect():
        if hast_to_run:
             self.tempdir = some_path
             self.config.executed_workflows.append(self.workflow_test.name)
```

I don't know if that breaks logging tho.

Reply:
> So, we create a temporary directory at instantiation of each workflow object, regardless of whether it's run.

No. When a workflow should not run it returns `[]` in the collector. Which means it is done collecting. Nothing happens. No temp dir creation.

> What about doing the temporary directory creation in the collect step? That would also obviate the need for keeping a self.config.workflow_cleanup_dirs variable around, as the directories to be removed can be looked up using an instance field or property from the workflows that were run.

Yeah that last bit happened when the teardown method was implemented. That was more elegant than keeping `self.config.workflow_cleanup_dirs` around. But unfortunately you cannot have custom tests that way. The cleanup has to be done in another stage and the cleanup directories have to be stored in some global variable that is accessible in that case (the config).

Reply:
aaah, nevermind me, I got confused here because the diff showed the entire tempdir creation as under YamlFile's `__init__` function.